mautic/mautic
# Dwell time query fix

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/2046
| BC breaks? | N
| Deprecations? | N

#### Description:
The dwell time dashboard widget does not display any data even though there are some. The reason was wrong order of dates in the `TIMESTAMPDIFF` MySql function so it returned negative values.

#### Steps to test this PR:
1. Apply this PR.
2. Refresh the page.

- The pie chart should appear if it has some values to show.

#### Steps to reproduce the bug:
1. Add the dwell time widget to your dashboard.

- There should not be any pie chart.
